packages/models/src/store/json-storage.ts

```diff
@@ -0,0 +1,87 @@
+import { ChainStorage, StorageOnChain, StorageOffChain } from './chain-storage'
+
+type JSONStorageType = string | number | boolean | bigint
+export type JSONStorageOffChain =
+  | JSONStorageType
+  | {
+      [x: string]: JSONStorageType | JSONStorageType[] | JSONStorageOffChain
+    }
+
+const typeMarkLen = 4
+const typeMarkMap: Record<string, string> = {
+  string: '0001',
+  number: '0010',
+  boolean: '0011',
+  bigint: '0100',
```

Comment:
Not sure if it's a good idea to introduce `number` and `bigint` simultaneously because precision in JavaScript leads to many troubles. How about using `big number` only?

Reply:
But only introducing `big number` would reduce the difficulty of manipulating numbers.

If `number` and `bigint` are used in a project, their drawbacks of them revealed in JavaScript ecosystem come too. In contrast, if only `big number` is there, users don't have to think about the precision or max safe number, etc.

It may have lower performance but has greater development experience.

Reply:
Actually, `number`'s performance is the same as `big number`. I think it's good to remove the `number` type.

Reply:
https://github.com/ckb-js/kuai/pull/51/commits/3045f1f3289e1f88478c6f01cee1433811207b5d